Adjustments to the iptables packetfilter implementation

tpantelis opened the pull request:
Minor adjustments and cleanup. See individual commits for details.

tpantelis committed:
Remove unused constants in iptables.go

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Store ipset.Interface instance in namedSet

...instead of the PacketFilter since it only needs the ipset.Interface.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
PacketFilter struct does not need to be exported

Rename to "packetFilter".

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Use maps for type to string conversions in iptables packetfilter

...instead of arrays. This makes the mappings more explicit and clearer
and removes the dependence on the ordering of the enums. This also
eliminates the need to define *MAX enums which actually aren't valid
types for user consumption anyway.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Simplify iptables ToRuleSpec

...by removing the return error. The only error returned was from the default
case of the RuleAction switch but we can change to a simple if/else to
avoid this.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Rename iptables packetFilter method receivers from 'a' to 'p'

...as 'a' doesn't seem to have any meaning in this context.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

yboaron commented on pkg/packetfilter/fake/packetfilter.go:
```diff
@@ -44,12 +44,12 @@ type PacketFilter struct {
 
 var (
-	iphookChainTypeToTableType = [packetfilter.ChainTypeMAX]packetfilter.TableType{
+	iphookChainTypeToTableType = []packetfilter.TableType{
```
maybe we should use map also here (same as your suggested change for pkg/packetfilter/iptables/iptables.go) ?

tpantelis replied:
I didn't change this one b/c I actually remove it with https://github.com/submariner-io/submariner/pull/2924 😄

yboaron commented on pkg/packetfilter/iptables/iptables.go:
```diff
@@ -327,9 +307,7 @@ func ToRuleSpec(rule *packetfilter.Rule) ([]string, error) {
 		ruleSpec = append(ruleSpec, "--set-mark", rule.MarkValue)
 	}
 
-	logger.Infof("ToRuleSpec: %s", strings.Join(ruleSpec, " "))
```
I think it would be useful to print ruleSpec here for debugging

tpantelis replied:
OK - we can leave it in as TRACE - I think as just Info level it might fill logs too much. Or perhaps we should log the spec when something fails...

yboaron replied:
Let's keep it as TRACE , to make sure we didn't miss something in packetfilter rule to iptables rule translation

tpantelis replied:
I pushed commit https://github.com/submariner-io/submariner/pull/2925/commits/a0c44f52a5d9021e1450ad9922269ec7a4a533c6.  I put the TRACE log back in and also added the rule spec string to the error logging.

tpantelis replied:
Also added a `String` method for `Rule`: https://github.com/submariner-io/submariner/pull/2925/commits/21e672b4c73a32d67d9ce62acca782a2ea91fec2.

tpantelis committed:
Adjust error messages in iptables packetfilter

Add more context, specifically the rule spec string, to the output. Also
re-add the logging in ToRuleSpec but as TRACE.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Add String method for packetfilter Rule

This outputs enum strings instaed of they're ordinal values. Also
omit fields that aren't set to trim the output as most fields won't
apply for all use cases.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Define RuleSpec type to wrap []string in iptables

...and define the String() method to wrap strings.Join. This avoids
calling strings.Join for formatting in error messages et al every time
on the fast path when formatting is elided.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

yboaron committed:
Merge branch 'devel' into iptables
